cpp/oneapi/dal/algo/pca/backend/cpu/finalize_train_kernel_cov.cpp

```diff
@@ -89,7 +89,9 @@ static train_result<Task> call_daal_kernel_finalize_train(const context_cpu& ctx
         interop::convert_to_daal_homogen_table(arr_cor_matrix, column_count, column_count);
     daal_cov::Parameter daal_parameter;
     daal_parameter.outputMatrixType = daal_cov::correlationMatrix;
-
+    if (desc.do_scale() == false && desc.do_mean_centering() == true) {
```

Comment:
Can you please explain what is going on with this condition? I would propose to use temporary variables to better express intention

Reply:
In general if both is true, it will mean that we must to compute Correlation matrix, if we need just mean_centering(it will mean that we do compute Covariance matrix), for aligning with sklearn

Reply:
And for sure, I will simplify it with separate variables